main.py

```diff
@@ -472,10 +480,31 @@ def generate_topography_maps(
             )
 
         # Step 4: Mire detection + detect meridinial points on respective mires
-        image_cent_list, center, others = process(
-            image_seg, image_gray, center, self.jump, self.start_angle, self.end_angle
-        )
-        _, _, image_mp = others
+        if (dl_segmentation_flag):
+            print("Running segmentation on mask...")
+            # Converting to three channels as expected by segment_mires
+            image_gray_copy = image_gray.copy()
+            image_gray_with_channels = np.dstack((image_gray.copy(), np.dstack((image_gray_copy, image_gray_copy))))
+            # generate mask
+            mire_mask = get_mask(image_gray_with_channels)
+            # dump mask
+            image_prefix = image_name.split(".jpg")[0]
+            mask_file_path = self.output + "/" + image_prefix + "/" + image_prefix + "_mire_mask.csv"
+            mire_mask = mire_mask.astype(int)
+            generate_colored_image(mire_mask, self.output + "/" + image_prefix + "/" + image_prefix + "_mire_mask.png")
+            np.savetxt(mask_file_path, mire_mask, delimiter=",", fmt="%d")
+            # segment mires

```

Comment:
I would move this logic to a function inside the segmentation_dl file. Call it detect_mires_in_image().

- Pass it the gray scale image and let it do the logic of stacking channels, etc.
- Also pass the output file name which it will write to.